train.py

```diff
@@ -0,0 +1,228 @@
+        PitchSeqNN: The trained PitchSeqNN model.
+
+    This function initializes an experiment on WandB, initializes the model, dataloaders and optimizer, and performs
+    the training and validation loops. It logs the training and validation statistics to WandB.
+    """
+    # loading data
+    dataset = BagOfPitches(selected_composers=classnames)
+    v_dataset = BagOfPitches(split="validation", selected_composers=classnames)
+    train_dataloader = DataLoader(dataset, batch_size=cfg.hyperparameters.batch_size, shuffle=True)
+    v_dataloader = DataLoader(v_dataset, batch_size=cfg.hyperparameters.batch_size, shuffle=True)
+
+    # initialize experiment on WandB with unique run id
+    initialize_wandb(cfg, classnames)
+
+    # initialize model, optimizer and loss criterion
+    model = PitchSeqNN(cfg.model.hidden_layers, 128, len(classnames))
+    optimizer = torch.optim.Adam(model.parameters(), lr=cfg.hyperparameters.lr)
+    criterion = nn.CrossEntropyLoss()
+
+    pbar = tqdm(range(cfg.hyperparameters.num_epochs), desc="Training started!")
+
+    # training loop
+    for epoch in pbar:
+        train_stats = training_epoch(
+            train_dataloader=train_dataloader,
+            optimizer=optimizer,
+            model=model,
+            criterion=criterion,
+        )
+
+        v_stats = validation_epoch(
+            loader=v_dataloader,
+            model=model,
+            criterion=criterion,
+        )
+
+        wandb.log({**train_stats, **v_stats})
+
+        bar = (
+            "loss={t_loss:.3f}, acc={t_acc:.2f}, f1_score={f1:.2f}, val_loss={v_loss:.3f}, val_acc={v_acc:.2f}, "
+            "val_f1_score={v_f1:.2f}"
+        ).format(
+            t_loss=train_stats["loss"],
+            t_acc=train_stats["accuracy"],
+            f1=train_stats["f1_score"],
+            v_loss=v_stats["val_loss"],
+            v_acc=v_stats["val_accuracy"],
+            v_f1=v_stats["val_f1_score"],
+        )
+        pbar.set_description(bar)
+    wandb.finish()
+    return model
+
+
+def training_epoch(
+    train_dataloader: DataLoader,
+    optimizer: torch.optim.Optimizer,
+    model: nn.Module,
+    criterion: Callable,
+):
+    correct = 0
+    running_loss = 0
+    truths = torch.tensor([])
+    preds = torch.tensor([])
+
+    for batch in train_dataloader:
+        inputs = batch["data"]
+        labels = batch["label"]
+
+        optimizer.zero_grad()
+
+        pred = model(inputs)
+        loss = criterion(pred, labels)
+        loss.backward()
+
+        preds = torch.concatenate((preds, pred.argmax(1)))
+        truths = torch.concatenate((truths, labels))
+        optimizer.step()
+
+        correct += (pred.argmax(1) == labels).sum().item()
+
+        running_loss += loss.item()
+
+    running_loss = running_loss / len(train_dataloader)
+    accuracy = correct / len(train_dataloader.dataset)
+    f1 = f1_score(truths, preds, average="weighted")
+    stats = {"loss": running_loss, "accuracy": accuracy, "f1_score": f1}
```

Comment:
```suggestion
    stats = {"train/loss": running_loss, "train/accuracy": accuracy, "train/f1_score": f1}
```

This will create a separate "train" section in the wandb dashboar, and it's very convenient to separate metrics from different splits.

You can also add this prefix after the epoch run, like:

```python
train_stats = {f"train/{key}": value for key, value in train_stats.items()}